Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,34 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python?(py)
+        requirements.any? do |req|
+          req.name == py && (req.required? || build.with?(req))
+        end ||
+          deps.any? do |dep|
+            dep.name == py && (dep.required? || build.with?(dep))
+          end
```

Comment:
looks like it would have to be `(requirements.to_a | deps)` or `(deps.to_a | requirements)` or `(requirements.to_a | deps.to_a)`